### PR TITLE
Add RV32 atomics zabha zacas

### PR DESCRIPTION
**Pull request by benoy-alexander**

### CODE REVIEW

File: .gitignore
Comment (benoy-alexander) [inline .gitignore:1]: These files/dirs get generated if you follow the README

File: impl/forms/ExtractorDerived.h [hunk 1]
```diff
@@ -3206,5 +3206,104 @@ class Extractor<Form_PF_hint> : public ExtractorBase<Form_I>
...
 
+/**
+ * Atomic Double which has reads a pair of memory locations and can
+ * write a pair of memory locations
+ * RV32 NOTE: rs1 is the address base, rd is the dest data starting register.
+ * The second register added will be rd+1 and rs2+1
+ * Also, checks for alignment of those registers.
+ */
+template<>
+class Extractor<Form_AMO_pair> : public Extractor<Form_AMO>
```
Comment (benoy-alexander) [inline impl/forms/ExtractorDerived.h:3217]: Cannot use multiple inheritance of the load & store pair Extractors which require virtual inheritance in the base classes used by many other extractors.  Talked with @klingaard and this is not what folks want to do right now.

File: json/isa_rv32zacas.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    "stencil" : "0x2800202f",
+    "type" : ["int", "load", "atomic"],
+    "l-oper" : "all",
+    "data" : 32
+  },
+  {
+    "mnemonic" : "amocas.d",
+    "tags" : ["zacas"],
+    "form" : "AMO",
+    "xform" : "AMO_pair",
```
Comment (benoy-alexander) [inline json/isa_rv32zacas.json:15]: This is BOTH a load pair and store pair instruction!

File: test/directed/main.cpp [hunk 3]
```diff
@@ -123,7 +125,7 @@ int main(int argc, char **argv)
         {
             isa_files.emplace_back("json/isa_rv32cf.json");
         }
-        mavis_facade.reset(new MavisType(isa_files, {"uarch/uarch_rv64g.json"}));
+        mavis_facade.reset(new MavisType(isa_files, {"uarch/uarch_rv32g.json"}));
```
Comment (benoy-alexander) [inline test/directed/main.cpp:128]: I think this was a benign bug but went ahead and fixed it

File: json/isa_rv32zabha.json
Comment (klingaard) [inline json/isa_rv32zabha.json:1]: Curious, what's the difference between all of these variants?  Yea, I could read the ISA spec, but, well, ya know.

**Review by klingaard — APPROVED** [review]
@bdutro to be made aware new extensions coming for his other PR

**Comment by bdutro** [comment]
> @bdutro to be made aware new extensions coming for his other PR

Thanks, I'll update it after this PR merges.

File: json/isa_rv32zabha.json
Comment (benoy-mips) [reply json/isa_rv32zabha.json:1]: These variants allow for smaller sizes of byte and halfword.  It also adds Compare And Swap (CAS) versions.

**Review by colbynyce-mips — APPROVED** [review]

**Pull request merged**